lib/queue.js

```diff
@@ -1,33 +1,54 @@
+// last in in first out
 class Queue {
     constructor() {
-        // this.store = ...
-        throw new Error("This method has not been implemented!");
+        this.store = new Array(20);
+        this.front = 0;
+        this.rear = 0;
+        this.size = 0;
+        this.bufferSize = 20
```

Comment:
You might think about making `bufferSize` a parameter that can be passed in and defaults to 20. This way we can create Queues with variable buffer size!
```suggestion
        this.bufferSize = bufferSize;
        this.store = new Array(bufferSize);
        this.front = 0;
        this.rear = 0;
        this.size = 0;
```